Traceback (most recent call last):
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Plot results
fig, ax = plt.subplots(2,2,figsize=(12,12))

Nz = 100
T, dz = heat_equation(20, 0, 5, Nz, 24, 3600)
ax[0,0].plot(T,-dz*np.arange(Nz));
ax[0,0].set_xlabel('Temperature [ÂºC]')
ax[0,0].set_ylabel('Depth [m]')

T, dz = heat_equation(20, 0, 5, Nz, 24*14, 3600)
ax[0,1].plot(T,-dz*np.arange(Nz));
ax[0,1].set_xlabel('Temperature [ÂºC]')
ax[0,1].set_ylabel('Depth [m]')

T, dz = heat_equation(20, 0, 5, Nz, 24*30, 3600)
ax[1,0].plot(T,-dz*np.arange(Nz));
ax[1,0].set_xlabel('Temperature [ÂºC]')
ax[1,0].set_ylabel('Depth [m]')

T, dz = heat_equation(20, 0, 5, Nz, 24*365, 3600)
ax[1,1].plot(T,-dz*np.arange(Nz));
ax[1,1].set_xlabel('Temperature [ÂºC]')
ax[1,1].set_ylabel('Depth [m]')

plt.show()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Input [0;32mIn [2][0m, in [0;36m<cell line: 5>[0;34m()[0m
[1;32m      2[0m fig, ax [38;5;241m=[39m plt[38;5;241m.[39msubplots([38;5;241m2[39m,[38;5;241m2[39m,figsize[38;5;241m=[39m([38;5;241m12[39m,[38;5;241m12[39m))
[1;32m      4[0m Nz [38;5;241m=[39m [38;5;241m100[39m
[0;32m----> 5[0m T, dz [38;5;241m=[39m [43mheat_equation[49m[43m([49m[38;5;241;43m20[39;49m[43m,[49m[43m [49m[38;5;241;43m0[39;49m[43m,[49m[43m [49m[38;5;241;43m5[39;49m[43m,[49m[43m [49m[43mNz[49m[43m,[49m[43m [49m[38;5;241;43m24[39;49m[43m,[49m[43m [49m[38;5;241;43m3600[39;49m[43m)[49m
[1;32m      6[0m ax[[38;5;241m0[39m,[38;5;241m0[39m][38;5;241m.[39mplot(T,[38;5;241m-[39mdz[38;5;241m*[39mnp[38;5;241m.[39marange(Nz));
[1;32m      7[0m ax[[38;5;241m0[39m,[38;5;241m0[39m][38;5;241m.[39mset_xlabel([38;5;124m'[39m[38;5;124mTemperature [ÂºC][39m[38;5;124m'[39m)

Input [0;32mIn [1][0m, in [0;36mheat_equation[0;34m(bc_surface, bc_bottom, depth, Nz, integration, dt)[0m
[1;32m      7[0m [38;5;124;03m''' Solves the heat equation[39;00m
[1;32m      8[0m [38;5;124;03mbc_surface :: boundary condition at the surface[39;00m
[1;32m      9[0m [38;5;124;03mbc_bottom  :: boundary condition at the bottom[39;00m
[0;32m   (...)[0m
[1;32m     13[0m [38;5;124;03mdt         :: time step [s][39;00m
[1;32m     14[0m [38;5;124;03m'''[39;00m
[1;32m     16[0m [38;5;66;03m# Definitions[39;00m
[0;32m---> 17[0m dz    [38;5;241m=[39m [43mb[49m[38;5;241m/[39mNz    [38;5;66;03m# Distance between grid points[39;00m
[1;32m     18[0m alpha [38;5;241m=[39m [38;5;241m1.2e-6[39m  [38;5;66;03m# Conductivity[39;00m
[1;32m     20[0m [38;5;66;03m# Initialize temperature and depth field[39;00m

[0;31mNameError[0m: name 'b' is not defined
NameError: name 'b' is not defined

